File: 1395. Count Number of Teams/count_number_of_teams.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        n = len(rating)
        teams = 0

        for i in range(1, n-1):
            less_left = more_left = less_right = more_right = 0

            for j in range(i):
                if rating[j] < rating[i]:
                    less_left += 1
                elif rating[j] > rating[i]:
                    more_left += 1

            for r in range(i+1, n):
                if rating[r] > rating[i]:
                    more_right += 1
                elif rating[r] < rating[i]:
                    less_right += 1
            teams += less_left * more_right + more_left * less_right

        return teams

File: 1395. Count Number of Teams/CountNumberOfTeams.java
class Solution {
    public int numTeams(int[] rating) {
        int n = rating.length;
        int teams = 0;
        TreeSet<Integer> left = new TreeSet<>();
        TreeSet<Integer> right = new TreeSet<>();

        for (int r : rating) {
            right.add(r);
        }

        for (int r : rating) {
            right.remove(r);
            int lessLeft = left.headSet(r).size();
            int moreLeft = left.size() - lessLeft;
            int lessRight = right.headSet(r).size();
            int moreRight = right.size() - lessRight;
            teams += lessLeft * moreRight + moreLeft * lessRight;
            left.add(r);
        }

        return teams;
    }
}
